tests/components/tasmota/test_config_flow.py

```diff
@@ -1,8 +1,47 @@
 """Test config flow."""
+from homeassistant.components.mqtt.models import Message
 
 from tests.common import MockConfigEntry
 
 
+async def test_mqtt_abort_if_existing_entry(hass, mqtt_mock):
+    """Check MQTT flow aborts when an entry already exist."""
+    MockConfigEntry(domain="tasmota").add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        "tasmota", context={"source": "mqtt"}
+    )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_mqtt_abort_invalid_topic(hass, mqtt_mock):
+    """Check MQTT flow aborts if discovery topic is invalid."""
+    discovery_info = Message("", "", 0, False, subscribed_topic="custom_prefix/##")
+    result = await hass.config_entries.flow.async_init(
+        "tasmota", context={"source": "mqtt"}, data=discovery_info
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "invalid_discovery_info"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6299130)